include/flatbuffers/flatbuffers.h

```diff
@@ -1097,12 +1109,14 @@ class FlatBufferBuilder {
         finished(false),
         minalign_(1),
         force_defaults_(false),
-        dedup_vtables_(true),
-        string_pool(nullptr) {
+        dedup_vtables_(true)
+        #ifndef FLATBUFFERS_NO_STL
+        , string_pool(nullptr)
```

Comment:
It would be better to define a dummy `StringOffsetMap` and use it for `string_pool` without excessive `ifndef` in code. For example, `void*` or a `struct` with `clear()` method.